### ISSUE TITLE
cannot plot large images

### ISSUE THREAD

**Issue by jvo203** [JuliaAstro/AstroImages.jl#10]
optical FITS, NAXIS = 2, dimensions 24900 x 26620, 2.5GB FITS file, the following code runs out of memory:

using AstroImages
#] pkg> add https://github.com/JuliaAstro/AstroImages.jl

using FITSIO
using Plots

dir = "/home/chris/ダウンロード"
file = "e20121211_0010500001_dp_sf_st_mos.fits"

fitspath = string(dir, "/", file)
println("loading ", fitspath)
fits = load(fitspath)

img = AstroImage(fits)
plot(img)

The FITS file can be downloaded from

http://surveys.roe.ac.uk/wsa/cgi-bin/siap.cgi?extNum=2&file=djoser:/disk39/wsa/products/mosaics/20121211_v10/e20121211_0010500001_dp_sf_st_mos.fit

Then the file extension needs to be changed from .fit to .fits .

**Comment by giordano** [comment]
Wow, that's a big one, thanks for the report!  I don't think there is much we can do soon, unfortunately.

I see that [Astropy uses memory-mapped FITS files](http://docs.astropy.org/en/v3.1.2/io/fits/appendix/faq.html#how-do-i-open-a-very-large-image-that-won-t-fit-in-memory) to deal with this issue.  We would need support for memmap in [`FITSIO.jl`](https://github.com/JuliaAstro/FITSIO.jl) as well.

**Comment by jvo203** [comment]
Just to clarify, AstroImages.jl as well as FITSIO.jl can load the file and println the data itself without a problem. It is the plotting routine that runs out of memory...

**Comment by giordano** [comment]
I tried several tools in Julia, all of them failed:

* Plots.jl: segmentation fault and large memory consumption
* GR.jl: this is probably broken on my system
* ImageView.jl: segmentation fault and large memory consumption
* PyPlot: I killed Julia because of large memory consumption
* Gadfly.jl: I killed Julia after it reached 23 GiB (!) of memory used
* Makie.jl (on master): it didn't eat all my memory up nor made Julia segfault, but it just errored:
   ```
   Error showing value of type Scene:
   ERROR: glTexImage 2D: width too large. Width: 24900
   ```

I must admit that for the time being there is no tool able to deal with data of this size.  The most promising is Makie.jl, but right now it just refuses to plot the image.

**Comment by timholy** [comment]
This should work automatically in ImageView 0.11.

**Comment by giordano** [comment]
Yes, I tried `imshow(rand(Float32, 24900, 26620))` and it was fairly smooth, thanks!

**Comment by timholy** [comment]
Great. The ability to zoom in to the pixel level should not be compromised, as the downsampling occurs just before the data get copied to the canvas (see https://github.com/JuliaImages/ImageView.jl/pull/265).

**Comment by sefffal** [comment]
Thanks to merging #30, this should now work in the plot recipes as well.
It just calls restrict as needed until it's a more reasonable size to plot.
